messages/src/message.rs

```diff
@@ -679,6 +717,132 @@ impl SerializedSize for Status {
     const MAX_SIZE: usize = core::mem::size_of::<Status>();
 }
 
+bitflags::bitflags! {
+    #[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
+    pub struct StatusV2: u32 {
+        /// The module is present in the receptacle.
+        ///
+        /// This translates to the `ModPrsL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.4.
+        const PRESENT               = 0b0000_0000_0001;
+
+        /// The module's power is enabled.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const ENABLED               = 0b0000_0000_0010;
+
+        /// The module is held in reset.
+        ///
+        /// This translates to the `ResetL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.2.
+        const RESET                 = 0b0000_0000_0100;
+
+        /// The module is held in low-power mode.
+        ///
+        /// This translates to the `ResetL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.3.
+        const LOW_POWER_MODE        = 0b0000_0000_1000;
+
+        /// At least one interrupt is signaled on the module. The details of the
+        /// interrupt cause can be queried by reading various bytes of the QSFP
+        /// memory map, such as bytes 3-21 in the lower memory page. See
+        /// SFF-8636 rev 2.10a section 6.2.3 for details.
+        ///
+        /// This translates to the `IntL` pin in SFF-8679 rev 1.8 section
+        /// 5.3.5.
+        const INTERRUPT             = 0b0000_0001_0000;
+
+        /// This module's power supply has come up successfully.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const POWER_GOOD            = 0b0000_0010_0000;
+
+        /// This module's power supply has not come up after being enabled.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const FAULT_POWER_TIMEOUT   = 0b0000_0100_0000;
+
+        /// This module unexpectedly lost power.
+        ///
+        /// Note that this is not part of the QSFP specification, but provided
+        /// by the Sidecar board design itself.
+        const FAULT_POWER_LOST      = 0b0000_1000_0000;
+
+        /// The SP has disabled the port associated with this module
+        ///
+        /// This occurs if the module was initialized but is now NACKing I2C
```

Comment:
This particular case is the one we've encountered so far, right? What I mean is, it seems like we should make clear this is an example, or we could make this specific case its own bit. E.g., `SP_DISABLED_DUE_TO_NACK` or something less terrible.

Reply:
Yeah, I clarified that this is just an example (although this happens to be the only reason the SP will disable a port _right now_).